specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/preview/2024-03-01-preview/diagnostic.json

```diff
@@ -0,0 +1,695 @@
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Returns spot VM placement scores for given configurations.": {
+            "$ref": "./examples/SpotPlacementRecommender.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ComputeDiagnosticsMetadata": {
+      "description": "Lists all available Compute diagnostics for a subscription in a location.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ComputeDiagnosticBase"
+          },
+          "description": "The collection of available Compute diagnostics returned by the listing operation."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The continuation token."
+        }
+      }
+    },
+    "ComputeDiagnosticBase": {
+      "description": "Contains metadata of a diagnostic type",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Identifier for the type of diagnostic",
+          "type": "string"
+        },
+        "name": {
+          "description": "Friendly name for the type of diagnostic",
+          "type": "string"
+        },
+        "type": {
+          "description": "Type of the diagnostic",
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/DiagnosticProperties"
+        }
+      }
+    },
+    "DiagnosticProperties": {
+      "description": "Contains additional properties of a diagnostic",
+      "type": "object",
+      "properties": {
+        "supportedResourceTypes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Describes what are the supported resource types for a diagnostic."
+        }
+      }
+    },
+    "RunDiskInspectionInput": {
+      "type": "object",
+      "properties": {
+        "resourceId": {
+          "description": "Qualified ID of the resource.",
+          "type": "string"
+        },
+        "manifest": {
+          "description": "Name of manifest in order to trigger Disk Inspection.",
+          "type": "string"
+        },
+        "uploadSasUri": {
+          "description": "SAS uri to the blob where results will be uploaded.",
+          "type": "string",
+          "x-ms-secret": true
+        }
+      },
+      "required": [
+        "resourceId",
+        "manifest",
+        "uploadSasUri"
+      ],
+      "description": "Data used for requesting a Disk Inspection execution."
+    },
+    "ComputeDiagnosticsOperationResult": {
+      "type": "object",
+      "properties": {
+        "message": {
+          "type": "string",
+          "description": "The result of the disk inspection operation."
```

Comment:
> disk inspectio [](http://example.com/codeflow?start=43&length=15)

is it always a disk inspection operation?

Reply:
Yes, this object is only returned in DiskInspection related operations